nvflare/apis/impl/wf_comm_server.py

```diff
@@ -134,7 +134,7 @@ def _set_stats(self, fl_ctx: FLContext):
                     "collector must be an instance of GroupInfoCollector, but got {}".format(type(collector))
                 )
             collector.set_info(
-                group_name=self._name,
+                group_name=self.controller._name,
```

Comment:
Does it make sense to pass in controller here though?

Reply:
Sure, we can look to change/improve `process_result_of_unknown_task()` which is implemented in several controllers, but used in the WFCommServer, so then we would no longer need to pass in controller here.

Reply:
We decided to keep `process_result_of_unknown_task()` for backwards compatibility, but added an event and removed the `@abstractmethod` decorator. For now `controller` will still need to be passed to `communicator` unless we remove `process_result_of_unknown_task()` in the future.